leetcode/src/main/java/com/wenmq/cn/leetcode/Solution1496.java

```diff
@@ -0,0 +1,45 @@
+package com.wenmq.cn.leetcode;
+
+import java.util.HashSet;
+
+/**
+ * Q1496
+ * https://leetcode-cn.com/problems/path-crossing/
+ */
+public class Solution1496 {
+
+    public static final int BIAS = 1 << 16;
+
+    /**
+     * 1 <= path.length <= 10^4
+     * path 仅由 {'N', 'S', 'E', 'W} 中的字符组成
+     */
+    public boolean isPathCrossing(String path) {
+        int position = 0;
+        HashSet<Integer> set = new HashSet<>(path.length());
```

Comment:
If the method return true, part of set memory won't be used. 
Passing no size parameter to HashSet will causes array copy continually. 
Anyway, there is always a weakness.